lib/charms/sdcore_nrf/v0/fiveg_nrf.py

```diff
@@ -0,0 +1,281 @@
+            return remote_app_relation_data.get("url")
+        return None
+
+    def _get_remote_app_relation_data(
+        self, relation: Optional[Relation] = None
+    ) -> Optional[Dict[str, str]]:
+        """Get relation data for the remote application.
+
+        Args:
+            Relation: Juju relation object (optional).
+
+        Returns:
+            Dict: Relation data for the remote application
+            or None if the relation data is invalid.
+        """
+        relation = relation or self.model.get_relation(self.relation_name)
+        if not relation:
+            logger.warning(f"No relation: {self.relation_name}")
+            return None
+        if not relation.app:
+            logger.warning("No remote application in relation: %s", self.relation_name)
+            return None
+        remote_app_relation_data = dict(relation.data[relation.app])
+        if not self._relation_data_is_valid(remote_app_relation_data):
+            logger.error("Invalid relation data: %s", remote_app_relation_data)
+            return None
+        return remote_app_relation_data
+
+    @staticmethod
+    def _relation_data_is_valid(relation_data: dict) -> bool:
+        """Returns whether URL is valid.
+
+        Args:
+            str: URL to be validated.
+
+        Returns:
+            bool: True if URL is valid, False otherwise.
+        """
+        try:
+            ProviderSchema(app=relation_data)
+            return True
+        except ValidationError:
+            return False
+
+
+class NRFProvides(Object):
+    """Class to be instantiated by the charm providing the NRF data."""
+
+    def __init__(self, charm: CharmBase, relation_name: str):
+        """Init."""
+        super().__init__(charm, relation_name)
+        self.relation_name = relation_name
+        self.charm = charm
+
+    @staticmethod
+    def _relation_data_is_valid(url: str) -> bool:
+        """Returns whether URL is valid.
+
+        Args:
+            str: URL to be validated.
+
+        Returns:
+            bool: True if URL is valid, False otherwise.
+        """
+        try:
+            ProviderSchema(app=ProviderAppData(url=url))  # type: ignore[arg-type]
+            return True
+        except ValidationError as e:
+            logger.error("Invalid url: %s", e)
+            return False
+
+    def set_nrf_information(self, url: str) -> None:
+        """Sets url in the application relation data.
+
+        Args:
+            str: NRF url
+            int: Relation ID
+
+        Returns:
+            None
+        """
+        if not self.charm.unit.is_leader():
+            raise RuntimeError("Unit must be leader to set application relation data.")
+        relations = self.model.relations[self.relation_name]
+        if not relations:
+            raise RuntimeError(f"Relation {self.relation_name} not created yet.")
+        if not self._relation_data_is_valid(url):
```

Comment:
_relation_data_is_valid takes the relation data as an input, not an url? And why are we checking that the existing relation data is valid?

Reply:
Fixed. I've changed the name of the method to `_url_is_valid` which is the actual functionality.

Reply:
I changed it accordingly to Ghislain's comment.